### PR TITLE
#169817874 new travel request notification

### PR DESCRIPTION
**Pull request by VedaDom**
#### What does this PR do?

This PR adds notification functionality.
When the requester request a new travel notification will be sent to the line manager for both in-app and via email.
#### Description of Task to be completed?

* There may be 2 types of notifications: email notifications and in-app notifications.
* Email notifications are sent as email messages to the email address the user submitted during registration.
* Users should be able to opt out (and in) of this type of notification.
* In-app notifications show up in the app. This may be visible on a notification page or some other view, as the team may choose.
#### How should this be manually tested?

- After clone repository, go to the root of the project open path into command run the `npm install` and once setup is done click `npm test`.

- In Postman, create a GET request with this url:`localhost:3000/api/v1/notifications` and pass `token` in headers
#### Screenshots

Screen Shot 2020-01-10 at 10 41 30

#### What are the relevant pivotal tracker stories?
- [#169817874](https://www.pivotaltracker.com/story/show/169817874)

### CODE REVIEW

**Comment by jo-rdan** [comment]
hey @VedaDom  am sorry but am a little bit confused, is this PR still in progress or it is ready?

**Review by mujohn26 — APPROVED** [review]

**Review by jo-rdan — CHANGES_REQUESTED** [review]
Dominque you did a wonderful job we will be using this in the future. however, I have left some comments you check them out. thanks

File: src/controllers/notification.controller.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * @class NotificationController
+ */
+class NotificationController {
+  /**
+   * Get notification for a user
+   * @param {Object} req The request object
+   * @param {Object} res The response object
+   * @returns {Promise} res
+   */
+  static async userNotifications(req, res) {
```
Comment (jo-rdan) [inline src/controllers/notification.controller.js]: I think we should name this method after the functionality that it provides for example _getUserNotification_

File: src/controllers/notification.controller.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    return response.successMessage(res, `${req.user.firstName}'s notifications`, 200, query);
+  }
+
+  /**
+   * View a spesific notification
+   * @param {Object} req The request object
+   * @param {Object} res The response object
+   * @returns {Promise} res
+   */
+  static async viewNotification(req, res) {
```
Comment (jo-rdan) [inline src/controllers/notification.controller.js]: this method should also be called after its functionality

**Review by jo-rdan — CHANGES_REQUESTED** [review]

File: src/controllers/notification.controller.js (hunk 1 above)
Comment (VedaDom) [reply src/controllers/notification.controller.js]: Your right on that, thanks.

**Review by jo-rdan — APPROVED** [review]

**Pull request merged**